stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -889,6 +913,13 @@ public static boolean isEntityDataSourceAvailable(BMap<String, BValue> struct) {
 
     private static void setCompressionHeaders(Context context, HttpCarbonMessage requestMsg, HttpCarbonMessage
             outboundResponseMsg) {
+        if (requestMsg.getHeaders().contains(SERVER_NAME)) {
```

Comment:
Shall we not use setCompressionHeaders method to set server header. We can have a separate method for that. WDYT?